File: MSVSimpleLeveliOS/MSVSimpleLeveliOS/ViewController.m
//
//  ViewController.m
//  MSVSimpleLeveliOS
//
//  Created by Serge Moskalenko on 20.09.16.
//  Copyright © 2016 Serge Moskalenko. All rights reserved.
//


#import "ViewController.h"
#import "Motions.h"

@interface ViewController ()
@property (nonatomic, strong) Motions* motions;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self runMotionManager];
}

- (void)viewWillDisappear:(BOOL)animated
{
    [self stopMotionManager];
}

- (BOOL)shouldAutorotate
{
    return NO;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Motion

- (void)runMotionManager
{
    _motions = [Motions new];
    [_motions runMotionManagerForLabel:self.labelView];
}

- (void)stopMotionManager
{
    [_motions stopMotionManager];
}

@end

File: MSVSimpleLeveliOS_swift/MSVSimpleLeveliOS/ViewController.swift
//
//  ViewController.h
//  MSVSimpleLeveliOS
//
//  Created by Serge Moskalenko on 20.09.16.
//  Copyright © 2016 Serge Moskalenko. All rights reserved.
//
import UIKit
class ViewController: UIViewController {
    @IBOutlet weak var labelView: UILabel!


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.runMotionManager()
    }

    override func viewWillDisappear(animated: Bool) {
        self.stopMotionManager()
    }

    override func shouldAutorotate() -> Bool {
        return false
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
//  - Motion -

    func runMotionManager() {
        self.motions = Motions()
        motions.runMotionManager(for: self.labelView)
    }

    func stopMotionManager() {
        motions.stopMotionManager()
    }

    var motions: Motions!
}
